packages/interfaces/src/message.ts

```diff
@@ -46,6 +46,10 @@ export interface IMetaSetter {
 }
 
 export interface EncoderOptions {
+  /**
+   * @deprecated
+   */
+  pubsubTopic?: PubsubTopic;
   pubsubTopicShardInfo?: SingleShardInfo;
```

Comment:
not a fan of this redundancy :(

Reply:
@danisharora099  This is temporary and a deprecated option. I kept it to allow autosharding support without introducing breaking changes for any existing users.

Another option would be to just keep `pubsubTopic` and make its type `pubsubTopic?: PubsubTopic | SingleShardInfo`, but I'm not sure how to communicate that we're deprecating part of a type. I'm open to other suggestions.

Reply:
I think deprecation period is a good way to go so that we will fully remove the flag in the upcoming update. 
In which case I think we should bump minor version and release `0.1.*`.